src/ToolInfo.php

```diff
@@ -23,83 +23,114 @@
+    {
+        if (!$this->isInstalledByComposer()) {
             throw new \LogicException('Cannot get composer version for tool not installed by composer.');
         }
 
-        if (null === $result) {
-            $composerInstalled = json_decode(file_get_contents(self::getScriptDir().self::COMPOSER_INSTALLED_FILE), true);
+        if (null === $this->composerVersion) {
+            $composerInstalled = json_decode(file_get_contents($this->getComposerInstalledFile()), true);
 
             foreach ($composerInstalled as $package) {
                 if (self::COMPOSER_PACKAGE_NAME === $package['name']) {
-                    $result = $package['version'].'#'.$package['dist']['reference'];
+                    $this->composerVersion = $package['version'].'#'.$package['dist']['reference'];
                     break;
                 }
             }
         }
 
-        return $result;
+        return $this->composerVersion;
     }
 
-    public static function getVersion()
+    public function getVersion()
     {
-        if (self::isInstalledByComposer()) {
-            return Application::VERSION.':'.self::getComposerVersion();
+        if ($this->isInstalledByComposer()) {
+            return Application::VERSION.':'.$this->getComposerVersion();
         }
 
         return Application::VERSION;
     }
 
-    public static function isInstalledAsPhar()
+    public function isInstalledAsPhar()
     {
-        static $result;
-
-        if (null === $result) {
-            $result = 'phar://' === substr(__DIR__, 0, 7);
+        if (null === $this->isInstalledAsPhar) {
+            $this->isInstalledAsPhar = 'phar://' === substr(__DIR__, 0, 7);
         }
 
-        return $result;
+        return $this->isInstalledAsPhar;
     }
 
-    public static function isInstalledByComposer()
+    public function isInstalledByComposer()
     {
-        static $result;
-
-        if (null === $result) {
-            $result = !self::isInstalledAsPhar() && file_exists(self::getScriptDir().self::COMPOSER_INSTALLED_FILE);
+        if (null === $this->isInstalledByComposer) {
+            $this->isInstalledByComposer = !$this->isInstalledAsPhar() && file_exists($this->getComposerInstalledFile());
         }
 
-        return $result;
+        return $this->isInstalledByComposer;
     }
 
-    private static function getScriptDir()
+    private function getComposerInstalledFile()
     {
-        static $result;
-
-        if (null === $result) {
-            $script = $_SERVER['SCRIPT_NAME'];
+        if (null === $this->composerInstalledFile) {
+            $script = $this->script;
+
+            while (is_link($script)) {
+                // On Windows, composer creates under vendor/bin executables that
+                // aren't symlinks, so we need to manually stop at a common point
+                // equal from Windows to Linux
+                if ('vendor'.DIRECTORY_SEPARATOR.'bin' === substr(dirname($script), -10)) {
```

Comment:
AFAIK, `DIRECTORY_SEPARATOR` isn't required as paths using `/` work on Windows with PHP.

Also, this will fail when Composer is configured to install dependencies in a different directory than `vendor`.